src/webview/js/js.js

```diff
@@ -49,6 +49,8 @@ import { toggleSpoiler } from './spoilers';
  */
 declare var platformOS: string;
 
+let doNotMarkMessagesAsReadFlag = false;
```

Comment:
I am using a global variable here since we need to use this in multiple functions inside `js.js`, it is defined within `handleInitialLoad` and used within `sendScrollMessage`, Is that appropriate? does there exists some better design pattern that I can follow here?

Reply:
Yeah, good question.

I think a somewhat better pattern would be to follow what we do for `platformOS`:
* It's defined directly in `script.js`, before the code in `compiledWebviewJs` (which has all the code in `js.js` and its friends) runs. This means it's initialized with the correct value from the beginning, and there's no need to worry about initialization order.
* Then there's a `declare var` line here at the top of `js.js`, so that Flow knows about it.

That's still kind of awkward, because there's no type-checking that what we do in `script.js` matches what `js.js` expects. But that's equally true with the parameters we pass to `handleInitialLoad`.

I think having it in a global is fine, so long as it has a nice explicit name like this. There's kind of an underlying messiness in the fact that this file is pretty long and does quite a lot of different things. So that makes it kind of uncomfortable to add more things to its top-level namespace. But I think the main thing for us to do there is to take some of the larger clusters of related code in this file and move them to separate modules.

Reply:
passing it like `platformOS` in new revision.